tests/ostree_mock.c

```diff
@@ -3,26 +3,135 @@
 #include <ostree.h>
 #include <stdio.h>
 
+/**
+ * @brief This mock allows the ostree package manager usage on non-ostree booted environment, e.g. local host
+ *
+ * in order to use it the following has to be pre-defined
+ * 1) OSTREE_DEPLOYMENT_VERSION_FILE environment variable that points to the test file containing a revision
+ * hash/checksum and its serial number 2) LD_PRELOAD environment variable that refers to the libostree_mock.so 3)
+ * [pacman].os == OSTREE_DEPLOYMENT_OS_NAME (aktualizr configuration, pacman section, os field must be equal to
+ * OSTREE_DEPLOYMENT_OS_NAME defined here)
+ *
+ * OSTREE_DEPLOYMENT_VERSION_FILE must follow this schema/format
+ *
+ * <deployed-revision-hash>\n
+ * <deployed-revision-serial>\n
+ * <booted-revision-hash>\n
+ * <booted-revision-serial>
+ *
+ * For example
+ *
+ *  d0ef921904bc573616a42e862de2421b265a76d06a334cfe5ad9f811107bbee0
+ *  0
+ *  d0ef921904bc573616a42e862de2421b265a76d06a334cfe5ad9f811107bbee0
+ *  0
+ */
+
+static const char OSTREE_DEPLOYMENT_OS_NAME[] = "dummy-os";
+static const char OSTREE_DEPLOYMENT_VERSION_FILE[] = "OSTREE_DEPLOYMENT_VERSION_FILE";
+
+static OstreeDeployment* ostree_sysroot_get_booted_deployment_from_file(const char* filename);
+
 OstreeDeployment* ostree_sysroot_get_booted_deployment(OstreeSysroot* self) {
-  OstreeDeployment* (*orig)(OstreeSysroot*) = dlsym(RTLD_NEXT, __func__);
-  char mocked_name[100];
-  snprintf(mocked_name, sizeof(mocked_name), "%s_mock", __func__);
-  OstreeDeployment* (*mocked)(OstreeSysroot*) = dlsym(RTLD_DEFAULT, mocked_name);
+  OstreeDeployment* (*orig)(OstreeSysroot*) = (OstreeDeployment * (*)(OstreeSysroot*))(dlsym(RTLD_NEXT, __func__));
+
+  const char* deployment_version_file = getenv(OSTREE_DEPLOYMENT_VERSION_FILE);
+  OstreeDeployment* deployment_from_file = NULL;
 
-  if (mocked) {
-    return mocked(self);
+  if (deployment_version_file != NULL &&
+      (deployment_from_file = ostree_sysroot_get_booted_deployment_from_file(deployment_version_file)) != NULL) {
+    return deployment_from_file;
   }
+
   return orig(self);
 }
 
-const char* ostree_deployment_get_bootcsum(OstreeDeployment* self) {
-  char* (*orig)(OstreeDeployment*) = dlsym(RTLD_NEXT, __func__);
-  char mocked_name[100];
-  snprintf(mocked_name, sizeof(mocked_name), "%s_mock", __func__);
-  char* (*mocked)(OstreeDeployment*) = dlsym(RTLD_DEFAULT, mocked_name);
+struct OstreeDeployment {
+  int serial;
+  char rev[255];
+};
+
+static int read_deployment_info(FILE* file, struct OstreeDeployment* deployment) {
+  struct OstreeDeployment deployment_res;
+
+  // TODO: make it more robust and reliable as a file might contain a value exceeding 255
+  if (fscanf(file, "%s\n", deployment_res.rev) != 1) {
```

Comment:
`"%254s\n`?

Reply:
Yeah, it should be better, although I don't like fscanf :), I think, it's good enough for the test component.